src/v0/destinations/salesforce/utils.js

```diff
@@ -47,59 +59,108 @@ const salesforceResponseHandler = (destResponse, sourceMessage, authKey, authori
-      // rudderJobMetadata contains some destination info which is being used to evict the cache
       ACCESS_TOKEN_CACHE.del(authKey);
       throw new RetryableError(
         `${DESTINATION} Request Failed - due to "INVALID_SESSION_ID", (Retryable) ${sourceMessage}`,
         500,
         destResponse,
       );
-    } else if (status === 403 && matchErrorCode('REQUEST_LIMIT_EXCEEDED')) {
-      // If the error code is REQUEST_LIMIT_EXCEEDED, you’ve exceeded API request limits in your org.
-      throw new ThrottledError(
-        `${DESTINATION} Request Failed - due to "REQUEST_LIMIT_EXCEEDED", (Throttled) ${sourceMessage}`,
+    default:
+      throw new AbortedError(
+        `${DESTINATION} Request Failed: "${status}" due to "${getErrorMessage(destResponse.response)}", (Aborted) ${sourceMessage}`,
+        400,
         destResponse,
       );
-    } else if (
-      status === 400 &&
-      matchErrorCode('CANNOT_INSERT_UPDATE_ACTIVATE_ENTITY') &&
-      (response?.message?.includes('UNABLE_TO_LOCK_ROW') ||
-        response?.message?.includes('Too many SOQL queries'))
-    ) {
-      // handling the error case where the record is locked by another background job
-      // this is a retryable error
+  }
+};
+
+const handleCommonAbortableError = (destResponse, sourceMessage, status) => {
+  throw new AbortedError(
+    `${DESTINATION} Request Failed: "${status}" due to "${getErrorMessage(destResponse.response)}", (Aborted) ${sourceMessage}`,
+    400,
+    destResponse,
+  );
+};
+
+/**
+ * ref: https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/errorcodes.htm
+ * handles Salesforce application level failures
+ * @param {*} destResponse
+ * @param {*} sourceMessage
+ * @param {*} stage
+ * @param {String} authKey
+ */
+const salesforceResponseHandler = (destResponse, sourceMessage, authKey, authorizationFlow) => {
+  const { status, response } = destResponse;
+
+  /**
+   *
+   * @param {*} errorCode
+   * response is of structure
+   * [
+   * {
+   *  "message" : "Request limit exceeded",
+   *  "errorCode" : "REQUEST_LIMIT_EXCEEDED"
+   * }
+   * ]
+   * @returns true if errorCode is found in the response
+   */
+  const matchErrorCode = (errorCode) =>
+    response && Array.isArray(response) && response.some((resp) => resp?.errorCode === errorCode);
+
+  switch (status) {
+    case 401:
+      if (authKey && matchErrorCode('INVALID_SESSION_ID')) {
+        handleAuthError(
+          'INVALID_SESSION_ID',
+          authKey,
+          authorizationFlow,
+          sourceMessage,
+          destResponse,
+          status,
+        );
+      }
+      handleAuthError('DEFAULT', authKey, authorizationFlow, sourceMessage, destResponse, status);
+      break;
+    case 403:
+      if (matchErrorCode('REQUEST_LIMIT_EXCEEDED')) {
+        throw new ThrottledError(
+          `${DESTINATION} Request Failed - due to "REQUEST_LIMIT_EXCEEDED", (Throttled) ${sourceMessage}`,
+          destResponse,
+        );
+      }
```

Comment:
```suggestion
      if (matchErrorCode('REQUEST_LIMIT_EXCEEDED')) {
        throw new ThrottledError(
          `${DESTINATION} Request Failed - due to "REQUEST_LIMIT_EXCEEDED", (Throttled) ${sourceMessage}`,
          destResponse,
        );
      }
      handleCommonAbortableError(destResponse, sourceMessage, status);
```
Should we change to this ?